### ISSUE TITLE
Function set membership not properly testing domain

### ISSUE THREAD

**Issue by thpani** [informalsystems/apalache#1735]
## Description

Apalache does not seem to properly check the function domain when testing membership with a function set.

## Input specification

```
--------------------- MODULE Test ---------------------
EXTENDS Integers
Init == TRUE
Next == TRUE
Inv ==
    LET f == [ x \in {2} |-> 2 ] IN
    LET S == [{ 1, 2 } -> { 2 }] IN
    f \in S
=======================================================
```

## The command line parameters used to run the tool

`apalache-mc check --inv=Inv ~/Test.tla`

## Expected behavior

A model-checking error.
TLC correctly reports the failing invariant.

## System information

- Apalache version [`apalache-mc version`]: `v0.25.1-3-g9f1aecbab`
- OS [e.g. Ubuntu Linux or Mac OS, and the current version]: macOS
- JDK version [e.g. OpenJDK 0.8.0]: Temurin-17.0.2+8

**Comment by Kukovec** [comment]
Adding to this, if you replace 
```tla
f \in S
```
with 
```tla
\E g \in S: g = f
```
The solver throws with
```sh
PASS #12: BoundedChecker                                          I@11:27:12.820
State 0: Checking 1 state invariants                              I@11:27:13.458
Trying to expand a complex set in \E g$2 \in ... Try to move \E out of a negated expression. Expression:  W@11:27:13.476
Unhandled exception                                               E@11:27:13.479
java.lang.UnsupportedOperationException: Expansion of FinFunSet[CellTFrom(Set(Int)), CellTFrom(Set(Int))] is not supported yet
	at at.forsyte.apalache.tla.bmcmt.rules.QuantRule.expandExistsOrForall(QuantRule.scala:121)
```
Which shouldn't be an `Unhandled exception` even if it's not supported.

**Comment by Kukovec** [comment]
Found the actual bug: in `SetInRule::funSetIn` the condition for `f \in [S -> T]` is described as
```
// In the new implementation, a function is a relation { <<x, f[x]>> : x \in U }.
// Check that \A t \in f: t[1] \in S /\ t[2] \in T.
``` 
Which implies `DOMAIN f \subseteq S`, but not `DOMAIN f = S`. I'll try to write a quick fix

**Comment by thpani** [comment]
> Found the actual bug: in `SetInRule::funSetIn` the condition for `f \in [S -> T]` is described as

Tagging for reference: this might be resolved by removing `SetInRule::funSetIn` in #1716

**Comment by thpani** [comment]
Reopening because this is still broken with `--smt-encoding=arrays`
